playground/signalr/SignalRServerlessWeb/Program.cs

```diff
@@ -1,21 +1,17 @@
 using Microsoft.Azure.SignalR.Management;
-using SignalRServerlessWeb;
 
 var builder = WebApplication.CreateBuilder(args);
 
-// Add services to the container.
-builder.Services.AddRazorPages();
-builder.Services.AddHttpClient();
-builder.Services.AddSingleton(sp =>
-{
-    return new ServiceManagerBuilder()
+var serviceManager = new ServiceManagerBuilder()
         .WithOptions(option =>
         {
             option.ConnectionString = builder.Configuration["ConnectionStrings:signalrServerless"];
         })
         .BuildServiceManager();
-});
-builder.Services.AddHostedService<BackgroundWorker>();
+var hubContext = await serviceManager.CreateHubContextAsync("myHubName", default);
+
+builder.Services.AddRazorPages();
+builder.Services.AddSingleton(hubContext).AddHostedService<PeriodicBroadcaster>();
```

Comment:
how it fetches ServiceHubContext in the ctor?

Reply:
So `.AddHostedService<PeriodicBroadcaster>()` registers `PeriodicBroadcaster` in the DI Container. And since the ctor of `PeriodicBroadcaster` has a dependency which is `ServiceHubContext`,  the DI Container fetches the `ServiceHubContext` singleton and pass it to the ctor when init the `PeriodicBroadcaster` instance

Reply:
why not directly add `new PeriodicBroadcaster(hubContext)` as the HostedService?

I am not sure if adding the ServiceHubContext is a best approach. this approach fails quickly when we have multiple hubs.

Reply:
that makes sense, just updated the code